src/txgraph.h

```diff
@@ -0,0 +1,212 @@
+     * before the C->B dependency is added, the dependency adding has no effect. If, together with
+     * the deletion of B also either A or C is deleted, there is no distinction between the
+     * original order case and the reordered case.
+     */
+    virtual void RemoveTransaction(const Ref& arg) noexcept = 0;
+    /** Add a dependency between two specified transactions. If a staging graph exists, the
+     *  dependency is only added there. Parent may not be a descendant of child already (but may
+     *  be an ancestor of it already, in which case this is a no-op). If either transaction is
+     *  already removed, this is a no-op. */
+    virtual void AddDependency(const Ref& parent, const Ref& child) noexcept = 0;
+    /** Modify the fee of the specified transaction, in both the main graph and the staging
+     *  graph if it exists. Wherever the transaction does not exist (or was removed), this has no
+     *  effect. */
+    virtual void SetTransactionFee(const Ref& arg, int64_t fee) noexcept = 0;
+
+    /** TxGraph is internally lazy, and will not compute many things until they are needed.
+     *  Calling DoWork will compute everything now, so that future operations are fast. This can be
+     *  invoked while oversized. */
+    virtual void DoWork() noexcept = 0;
+
+    /** Create a staging graph (which cannot exist already). This acts as if a full copy of
+     *  the transaction graph is made, upon which further modifications are made. This copy can
+     *  be inspected, and then either discarded, or the main graph can be replaced by it by
+     *  commiting it. */
+    virtual void StartStaging() noexcept = 0;
+    /** Discard the existing active staging graph (which must exist). */
+    virtual void AbortStaging() noexcept = 0;
+    /** Replace the main graph with the staging graph (which must exist). */
+    virtual void CommitStaging() noexcept = 0;
+    /** Check whether a staging graph exists. */
+    virtual bool HaveStaging() const noexcept = 0;
+
+    /** Determine whether the graph is oversized (contains a connected component of more than the
+     *  configured maximum cluster count). If main_only is false and a staging graph exists, it is
+     *  queried; otherwise the main graph is queried. Some of the functions below are not available
+     *  for oversized graphs. The mutators above are always available. Removing a transaction by
+     *  destroying its Ref while staging exists will not clear main's oversizedness until staging
+     *  is aborted or committed. */
+    virtual bool IsOversized(bool main_only = false) noexcept = 0;
+    /** Get the feerate of the chunk which transaction arg is in the main graph. Returns the empty
+     *  FeePerWeight if arg does not exist in the main graph. The main graph must not be
+     *  oversized. */
+    virtual bool Exists(const Ref& arg, bool main_only = false) noexcept = 0;
```

Comment:
Looks like comments for Exists/GetMainChunkFeerate/GetIndividualFeerate have been mixed up (in the "Add staging support" commit, I think)